keras/applications/convnext.py

```diff
@@ -0,0 +1,558 @@
+  else:
+    if not backend.is_keras_tensor(input_tensor):
+      img_input = layers.Input(tensor=input_tensor, shape=input_shape)
+    else:
+      img_input = input_tensor
+
+  if input_tensor is not None:
+    inputs = layer_utils.get_source_inputs(input_tensor)
+  else:
+    inputs = img_input
+
+  x = inputs
+  if include_preprocessing:
+    x = PreStem(name=model_name)(x)
+  
+  # Stem block.
+  stem = tf.keras.Sequential(
+    [
+      layers.Conv2D(projection_dims[0], kernel_size=4, strides=4, name=model_name + "_stem_conv"),
+      layers.LayerNormalization(epsilon=1e-6, name=model_name + "_stem_layernorm"),
+    ],
+    name=model_name + "_stem",
+  )
+
+  # Downsampling blocks.
+  downsample_layers = []
+  downsample_layers.append(stem)
+  for i in range(3):
+    downsample_layer = tf.keras.Sequential(
+        [
+          layers.LayerNormalization(epsilon=1e-6, name=model_name + "_downsampling_layernorm_" + str(i+1)),
+          layers.Conv2D(projection_dims[i + 1], kernel_size=2, strides=2, name=model_name + "_downsampling_conv_" + str(i+1)),
+        ],
+        name=model_name + "_downsampling_block_" + str(i+1),
+    )
+    downsample_layers.append(downsample_layer)
+  
+  # Stochastic depth schedule.
+  dp_rates = [x for x in tf.linspace(0.0, drop_path_rate, sum(depths))]
+
+  # ConvNeXt stages.
+  stages = []
+  cur = 0
+  for i in range(4):
+    stage = tf.keras.Sequential(
+      [
+        *[
+            Block(
+              projection_dim=projection_dims[i],
+              drop_path=dp_rates[cur + j],
+              layer_scale_init_value=layer_scale_init_value,
+              name=model_name + f"stage_{i}_block_{j}",
+            )
+            for j in range(depths[i])
+        ]
+      ],
+      name=model_name + f"_stage_{i}",
+    )
+    stages.append(stage)
+    cur += depths[i]
+
+  # Apply the stages.
+  for i in range(len(stages)):
+    x = downsample_layers[i](x)
+    x = stages[i](x)
+
+  if include_top:
+    x = Head(num_classes=classes)(x)
+    imagenet_utils.validate_activation(classifier_activation, weights)
+
+  else:
+    if pooling == "avg":
+      x = layers.GlobalAveragePooling2D()(x)
+    elif pooling == "max":
+      x = layers.GlobalMaxPooling2D()(x)
+      x = layers.LayerNormalization(epsilon=1e-6)(x)
+
+  model = training.Model(inputs=inputs, outputs=x, name=model_name)
+
+  # Load weights.
+  if weights == "imagenet":
+    if include_top:
+      file_suffix = ".h5"
+      file_hash = WEIGHTS_HASHES[model_name[-4:]][0]
+    else:
+      file_suffix = "_notop.h5"
+      file_hash = WEIGHTS_HASHES[model_name[-4:]][1]
+    file_name = model_name + file_suffix
+    weights_path = data_utils.get_file(
+        file_name,
+        BASE_WEIGHTS_PATH + file_name,
+        cache_subdir="models",
+        file_hash=file_hash)
+    model.load_weights(weights_path)
+  elif weights is not None:
+    model.load_weights(weights)
+
+  return model
+
+
+## Instantiating variants ##
+
+
+## TODO
```

Comment:
?

Reply:
Should be fixed now. It was placeholder.